### ISSUE TITLE
Issue with docs or my code?

### ISSUE THREAD

**Issue by justinrmiller** [RedisLabs/spark-redis#154]
I'm trying to build an example project for consuming Redis streams via dstreams. The code I have below is verbatim out of the docs, but it's not compiling. What am I missing? Is it something to do with the @transient annotation on RedisContext?

import com.redislabs.provider.redis._
import com.redislabs.provider.redis.streaming._

Don't appear to be used. 

Thanks! 
Justin

[error] /Users/justin/Development/Projects/spark-redis-example/src/main/scala/com/justinrmiller/sparkstreaming/Main.scala:19: value createRedisXStream is not a member of org.apache.spark.streaming.StreamingContext
[error]     val stream = ssc.createRedisXStream(Seq(ConsumerConfig("my-stream", "my-consumer-group", "my-consumer-1")))
[error]                      ^
[error] /Users/justin/Development/Projects/spark-redis-example/src/main/scala/com/justinrmiller/sparkstreaming/Main.scala:19: not found: value ConsumerConfig
[error]     val stream = ssc.createRedisXStream(Seq(ConsumerConfig("my-stream", "my-consumer-group", "my-consumer-1")))
[error]                                  

```
package com.justinrmiller.sparkstreaming

import com.redislabs.provider.redis._
import com.redislabs.provider.redis.streaming._
import org.apache.spark.sql.SparkSession
import org.apache.spark.streaming.{Seconds, StreamingContext}

object Main {

  def main(args: Array[String]) {
    val spark = SparkSession.builder
      .master("local[*]")
      .config("spark.host", "spark.redis.host")
      .config("spark.redis.port", "6379")
      .getOrCreate()

    val ssc = new StreamingContext(spark.sparkContext, Seconds(1))

    val stream = ssc.createRedisXStream(Seq(ConsumerConfig("my-stream", "my-consumer-group", "my-consumer-1")))
    stream.print()

    ssc.start()
    ssc.awaitTermination()
  }
}
```

**Comment by justinrmiller** [comment]
Full repo here: https://github.com/justinrmiller/spark-redis-example

**Comment by fe2s** [comment]
Hi @justinrmiller ,

`createRedisXStream` was introduced in spark-redis 2.3.1, so it's not available in 2.3.0. I agree it might be misleading why a new feature was added in a minor version. The versioning of spark-redis is different from common approach, the version of spark-redis is driven by the spark version that it's built against. At this point we should probably update documentation indicating which version the feature is available from.

To fix the compilation error you have to use version 2.3.1. Please also note, there is SBT issue resolving snapshot jedis dependency, so you have declare it like this

```
libraryDependencies ++= Seq(
  "com.typesafe"      % "config"                          % "1.3.0",
  "ch.qos.logback"    % "logback-classic"                 % "1.1.3",

  "org.apache.spark"  % "spark-streaming_2.11"            % sparkVersion,
  "org.apache.spark"  % "spark-sql_2.11"                  % sparkVersion,

  "com.redislabs" % "spark-redis" % "2.3.1" exclude("com.redislabs", "jedis"),
  "com.redislabs" % "jedis" % "3.0.0-20181113.105826-9" from "https://oss.sonatype.org/content/repositories/snapshots/com/redislabs/jedis/3.0.0-SNAPSHOT/jedis-3.0.0-20181113.105826-9.jar"
)
```

**Comment by justinrmiller** [comment]
That got the example working, thanks! Closing out.
